addons/mail/static/src/models/thread/thread.js

```diff
@@ -1401,6 +1401,22 @@ function factory(dependencies) {
             );
         }
 
+        /**
+         * Compute an url string that can be used inside a href attribute
+         *
+         * @private
+         * @returns {string}
+         */
+        _computeUrl() {
+            if (this.model !== 'mail.channel') {
+                const baseHREF = this.env.session.url('/web');
+                return `${baseHREF}#model=${this.model}&id=${this.id}`;
+            } else {
+                const baseHREF = this.env.session.url('/web');
+                return `${baseHREF}#action=mail.action_discuss&active_id=${this.model}_${this.id}&menu_id=${this.env.messaging.discuss.menu_id}`;
```

Comment:
Don't think you need menu id if you have `action=mail.action_discuss`, which should be enough.
If so, can remove field `discuss` and `discussMenuId` altogether.